precompiles/staking/integration_test.go

```diff
@@ -406,6 +406,69 @@ var _ = Describe("Calling staking precompile directly", func() {
 		})
 	})
 
+	Describe("to create validator", func() {
+		var (
+			defaultDescription = staking.Description{
+				Moniker:         "new node",
+				Identity:        "",
+				Website:         "",
+				SecurityContact: "",
+				Details:         "",
+			}
+			defaultCommission = staking.Commission{
+				Rate:          big.NewInt(100000000000000000),
+				MaxRate:       big.NewInt(100000000000000000),
+				MaxChangeRate: big.NewInt(100000000000000000),
+			}
+			defaultMinSelfDelegation = big.NewInt(1)
+			defaultPubkeyBase64Str   = "nfJ0axJC9dhta1MAE1EBFaVdxxkYzxYrBaHuJVjG//M="
+			defaultValue             = big.NewInt(1)
+
+			// defaultCreateValidatorArgs are the default arguments for the createValidator call
+			//
+			// NOTE: this has to be populated in the BeforeEach block because the private key is not initialized before
+			defaultCreateValidatorArgs contracts.CallArgs
+		)
+
+		BeforeEach(func() {
+			// populate the default createValidator args
+			defaultCreateValidatorArgs = defaultCallArgs.WithMethodName(staking.CreateValidatorMethod)
+		})
+
+		Context("validator address is the origin", func() {
+			It("should create validator success", func() {
+				createValidatorArgs := defaultCreateValidatorArgs.WithArgs(
+					defaultDescription, defaultCommission, defaultMinSelfDelegation, s.address, defaultPubkeyBase64Str, defaultValue,
+				)
+
+				logCheckArgs := passCheck.WithExpEvents(staking.EventTypeCreateValidator)
+
+				_, _, err := contracts.CallContractAndCheckLogs(s.ctx, s.app, createValidatorArgs, logCheckArgs)
+				Expect(err).To(BeNil(), "error while calling the smart contract: %v", err)
+
+				_, found := s.app.StakingKeeper.GetValidator(s.ctx, s.address.Bytes())
+				Expect(found).To(BeTrue(), "expected validator to be found")
+			})
+		})
+
+		Context("validator address is not the origin", func() {
```

Comment:
Nit: can you please add a test case that creates a validator using a smart contract that calls the precompile?

Reply:
@luchenqun please add the test cases, similar to the ones added in https://github.com/evmos/evmos/pull/2576/
The `StakingCaller.sol` in that PR includes the `testCreateValidator` function to use for the missing tests